activesupport/lib/active_support/core_ext/name_error.rb

```diff
@@ -1,6 +1,13 @@
 # frozen_string_literal: true
 
 class NameError
+  attr_writer :receiver
+  alias_method :original_receiver, :receiver
```

Comment:
How about prepending a module to avoid adding `original_receiver` to `NameError`'s API?